spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -193,28 +194,32 @@ public void destroy() throws Exception { //NOSONAR
 
 	@Override
 	public void start() {
-		this.running = true;
+		this.running.compareAndSet(false, true);
 	}
 
 
 	@Override
 	public void stop() {
-		try {
-			destroy();
-		}
-		catch (Exception e) {
-			logger.error("Exception while closing producer", e);
+		if (this.running.compareAndSet(true, false)) {
+			try {
+				destroy();
+			}
+			catch (Exception e) {
+				logger.error("Exception while closing producer", e);
+			}
 		}
 	}
 
 
 	@Override
 	public boolean isRunning() {
-		return this.running;
+		return this.running.get();
 	}
 
 	@Override
 	public Producer<K, V> createProducer() {
+		if (!isRunning())
```

Comment:
I don't think we need it here. Even if we don't start this component, we are going to clean up all the states in the `destroy()`, which is only a logic in the `stop()` implementation above.

I even wonder from here if we need the `Lifecycle` implementation here at all...

I don't tell that we need to change the logic right now, but let's figure out if it is obsolete at all!

Reply:
I'm also confused why it implemented `Lifecycle`, and if we implemented it, we must check running state before any operation.